registry/auth.go

```diff
@@ -215,7 +253,7 @@ func (err PingResponseError) Error() string {
 // challenge manager for the supported authentication types and
 // whether v2 was confirmed by the response. If a response is received but
 // cannot be interpreted a PingResponseError will be returned.
-func PingV2Registry(endpoint APIEndpoint, transport http.RoundTripper) (auth.ChallengeManager, bool, error) {
+func PingV2Registry(endpoint *url.URL, transport http.RoundTripper) (auth.ChallengeManager, bool, error) {
```

Comment:
This does not increase our reliance on the ping protocol. Ping is used here the same way it is used for any v2 endpoint, to discover the token server. If we figure out a different way to do discovery or start caching this information, we will be able to take advantage of that here as well.